src/VASSAL/build/module/map/MovementReporter.java

```diff
@@ -119,9 +120,8 @@ public Command markMoved(GamePiece p) {
     Command c = null;
     if (p instanceof Stack) {
       c = new NullCommand();
-      for (Iterator<GamePiece> i = ((Stack) p).getPiecesIterator();
-           i.hasNext();) {
-        c.append(markMoved(i.next()));
+      for (GamePiece gp : ((Stack)p).asList()) {
+        c.append(markMoved(gp));
```

Comment:
@BrentEaston Is this a bug? Should it be `c = c.append(markMoved(gp));`?

Reply:
Yes it should, otherwise the initial NullCommand is not replaced by the first append.

Reply:
Thanks. Fixed in 7dcc5ddda.